app/src/org/commcare/activities/CommCareSetupActivity.java

```diff
@@ -290,6 +292,12 @@ protected void onResume() {
             setResult(RESULT_OK);
             finish();
         }
+
+        // if we were in dependency dialog state and resumed CC, take the next step
+        // that we otherwise would have taken on a successful app install
+        if(uiState.equals(UiState.DEPENDENCY_DIALOG)){
+            launchNextActivityOnAppInstall();

```

Comment:
Out of curiosity does this end up retaining the "multiple" state when the previous user action transitioned the number of '"pending" dependencies from 2 to 1? I don't see a way for it to do so. I'm not sure what the cleanest way to retain that state is, but it seems like it would regress to the other behavior in between.

Reply:
that's right. Today we only show the dialog once after app install. I did that since I didn't want to block user indefinitely at this stage as there are other possible app manager functions that user might want to access to. We can probably check for the no. of unsatisfied apps here again and if the number decreased show the dialog again ?